### PR TITLE
feat: Integration with JIRA projects

### LINKED ISSUE #53: Add JIRA support to the bot
At the moment the snowdrop bot is able to connect to JIRA.

We should definitely include JIRA issues in the report and also display them in the `Issues` tab. 
From there we should consider other integrations too (maybe a bridge similar to the one we use for 3rd part repositories).

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Gathering of JIRA issues from the defined projects using a similar process as used in the github process.

### CODE REVIEW

File: src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  // These dates represent current reporting period
+  private final ZonedDateTime startTime;
+  private final ZonedDateTime endTime;
+
+  // These represent the oldest reporting period possible
+  private final Date minStartTime;
+  private final Date minEndTime;
+
+  public JiraIssueCollector(JiraRestClient client, StatusLogger status, int reportingDay, int reportingHour, Set<String> users, Set<String> repositories) {
+    this.client = client;
```
Comment (iocanel) [inline src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java:47]: Thanks for adopting the `this.` style!

File: src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+//    .map(status.<String>log(total, "Collecting issues from repository %s."))
+    .flatMap(i -> teamIssueStream(i));
+  }
+
+  private Stream<Issue> teamIssueStream(String repository) {
+    synchronized (client) {
+      LOGGER.info("Getting issues for repository {} and users {}", repository, users);
+      return StreamSupport.stream(
+      client.getSearchClient().searchJql("project = " + repository + " and updatedDate >= '" + DF.format(minStartTime) + "'").claim().getIssues()
+      .spliterator(), false).peek(i -> log(i))
```
Comment (iocanel) [inline src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java:104]: You're becoming a lambda jedi!

**Review by iocanel — APPROVED** [review]
Great work!

File: src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java (hunk 2 above)
Comment (jacobdotcosta) [reply src/main/java/io/snowdrop/jira/reporting/JiraIssueCollector.java:104]: more a copy+paste master...

**Pull request merged**